rust/agents/relayer/src/relayer.rs

```diff
@@ -30,6 +30,12 @@ use crate::{
     settings::{matching_list::MatchingList, GasPaymentEnforcementPolicy, RelayerSettings},
 };
 
+#[derive(Debug)]
+pub struct GasPaymentEnforcement {
```

Comment:
Okay, we need better names at this point...

Right now if I am following correctly we have:
- `GasPaymentEnforcementConfig`: Raw config that defines the rules on gay payments (whitelist + gas payment policy)
- `GasPaymentEnforcementPolicy`: Raw config that defines only the gas payment policy
- `GasPaymentEnforcement`: Partially parsed value that holds the raw config `GasPaymentEnforcementPolicy` and the parsed `MatchingList`
- `GasPaymentEnforcer`: fully parsed including `GasPaymentPolicy` and `MatchingList`, useful for ensuing a payment is sufficient given a message
- `GasPaymentPolicy`: Trait defining rules on how to decide if payments requirements have been met

IDK about you but this took me a while to figure out and was far from intuitive. I also don't think we should have the `GasPaymentEnforcement` if at all possible because it feels really weird to me having a partially deserialized type.

Reply:
yeah great callout, I just removed `GasPaymentEnforcement` like you suggested. Agree it feels pretty messy, lmk if you think there's room to make it more intuitive